messaging/src/main/java/io/atomix/messaging/impl/NettyMessagingService.java

```diff
@@ -545,9 +546,12 @@ private Bootstrap bootstrapClient(Endpoint endpoint) {
   private CompletableFuture<Void> startAcceptingConnections() {
     CompletableFuture<Void> future = new CompletableFuture<>();
     ServerBootstrap b = new ServerBootstrap();
+    b.option(ChannelOption.SO_REUSEADDR, true);
+    b.option(ChannelOption.SO_BACKLOG, 128);
     b.childOption(ChannelOption.WRITE_BUFFER_WATER_MARK,
-        new WriteBufferWaterMark(8 * 1024, 32 * 1024));
-    b.option(ChannelOption.SO_RCVBUF, 1048576);
```

Comment:
I believe SO_RCVBUF should be set on the child option, not option for server bootstrap.